R/between.R

```diff
@@ -47,7 +47,8 @@ between = function(x, lower, upper, incbounds=TRUE, NAbounds=TRUE, check=FALSE)
     if (isTRUE(getOption("datatable.verbose"))) cat("optimised between not available for this data type, fallback to slow R routine\n")
     if (isTRUE(NAbounds) && (anyNA(lower) || anyNA(upper))) stop("Not yet implemented NAbounds=TRUE for this non-numeric and non-character type")
     if (check && any(lower>upper, na.rm=TRUE)) stop("Some lower>upper for this non-numeric and non-character type")
-    if (incbounds) x>=lower & x<=upper
+    if (incbounds)
+      x>=lower & x<=upper
```

Comment:
what is this change about?

Reply:
just so that the simple regex for detecting `&` doesn't match this line.

could also restore this as eventually the linter would be able to distinguish this without the whitespace "crutch"

Reply:
so such kind of linter will not detect new pieces of code like this that will get introduced over time

Reply:
why not? it would be looking for things like

```
sub %iscall% 'if' && sub[[2L]] %iscall% c('&', '|')
```

the regex version would miss new similar lines of course, if that's what you mean

Reply:
I put the line back and used a comment instead to mark that this `&` is correct. That way the comment is displayed in the output from the new `grep` in CRAN_Release.cmd too.  Other `grep`'s there deploy the same strategy; i.e. where the grep show false positives, I add a comment to the line so it's marked.
The grep currently outputs : 
```
$ grep -nr "^[^#]*if[^&#]*[^&#\"][&][^&]" R | grep -Ev "if\s*[(](?:any|all)"
R/data.table.R:2558:    if (any(w <- new==names(x)[i] & Encoding(new)==Encoding(names(x)[i]))) {  # this & is correct not &&
R/between.R:50:    if (incbounds) x>=lower & x<=upper  # this & is correct not &&
```

The other reason I moved this line back was then I vertically aligned the two lines so we can instantly see the difference between them by comparing vertically.